backend/src/llm_client/base.py

```diff
@@ -0,0 +1,73 @@
+import abc
+import json
+from src.crawler.exceptions import DomainMismatchException
+from pydantic import BaseModel, Field
+
+class Message(BaseModel):
+    role: str = Field(..., example="user", description="The role of the sender (system, user, assistant)")
+    content: str = Field(..., example="Hello, how can I help you?", description="The content of the message")
+
+class LLMClientBase(metaclass=abc.ABCMeta):
+    
+    @abc.abstractmethod
+    def sum_up_news() -> dict:
+        
+        return NotImplemented
+    
+    @abc.abstractmethod
+    def extract_keywords() -> str:
+        
+        return NotImplemented
+    
+    @abc.abstractmethod
+    def evaluate_relevance() -> str:
+        
+        return NotImplemented
```

Comment:
It seems to me that an abstract method generating LLM text completion should be included here, and then provide implementation in the template.

```py
class LLMClientBase(metaclass=abc.ABCMeta):
    @abc.abstractmethod
    def _perform_request(self, ...) -> str:
        ...
```